### PR TITLE
e2e create-delete client interaction (DEV-2486)

### PR DESCRIPTION
**Pull request by tglaz**
### create-delete interaction e2e flow
DEV-2486

## Summary by Sourcery

Add an end-to-end flow for creating and deleting a client interaction, backed by reusable Maestro test flows and more robust, deterministic test infrastructure.

New Features:
- Introduce a Maestro e2e scenario that creates and then deletes a client interaction for a fixture client.
- Add reusable Maestro flows for launching the app with permissions, logging in, and visiting or creating a fixture client profile from the clients list.

Bug Fixes:
- Ensure the clients search query is reset whenever the Clients screen regains focus to avoid stale filters during navigation.
- Stabilize location-dependent interaction flows by pre-granting location permissions and setting a deterministic GPS location in tests.

Enhancements:
- Standardize and document the Maestro test directory layout, splitting runnable tests from reusable flows grouped by precondition kind.
- Add consistent, semantically meaningful testIDs across key client, interaction, note, and profile UI components to support e2e automation.
- Introduce a toTestId utility and migrate several ad hoc testID constructions to use it for predictable, collision-free identifiers.
- Expose additional props and passthrough behavior (e.g., testId on FieldCard and TextButton, rest props on IconButton) to improve testability and reuse.
- Tag client sub-screens and tabs with screen-level testIDs to make client detail flows easier to target in tests.
- Temporarily disable accessibility props on FieldCard pending a dedicated a11y compliance update and temporarily stub out the logout Maestro test flow.

Tests:
- Add unit tests for the toTestId utility covering normalization, joining, and edge cases.

### CODE REVIEW

File: libs/expo/betterangels/src/lib/screens/NoteForm/DeleteNote.tsx [hunk 1]
```diff
@@ -32,6 +32,7 @@ export default function DeleteNote(props: TDeleteNoteProps) {
         button={
           <Pressable
             disabled={isDeleting}
+            testID="delete-interaction-btn"
```
Comment (mikefeldberg) [inline libs/expo/betterangels/src/lib/screens/NoteForm/DeleteNote.tsx]: maybe `delete-note-btn` since all the other test ids use `note`?

File: libs/expo/betterangels/src/lib/screens/Client/Interactions/index.tsx [hunk 2]
```diff
@@ -44,15 +44,15 @@ export default function Interactions({
...
+    [client],
   );
 
   if (!client) {
     return null;
   }
 
   return (
-    <View style={styles.container}>
+    <View style={styles.container} testID="client-view-interactions">
```
Comment (mikefeldberg) [inline libs/expo/betterangels/src/lib/screens/Client/Interactions/index.tsx:55]: `client-view-notes`?

File: apps/betterangels/.maestro/README.md [hunk 3]
```diff
@@ -246,14 +246,30 @@ maestro --device <DEVICE_ID> test apps/betterangels/.maestro/tests \
...
 ```
 .maestro/
-  tests/
-    landing.yml          # Test flows
-
-  steps/
-    (reusable steps)
+  tests/                    # Runnable test scenarios (what `maestro test` runs)
+    client_crud.yml
+    client_interaction.yml
```
Comment (mikefeldberg) [inline apps/betterangels/.maestro/README.md]: `create_delete_client_interaction.yml`

File: apps/betterangels/.maestro/lib/screens/clients/search_and_select_client.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+appId: ${MAESTRO_APP_ID}
+---
+# Precondition: on the clients screen
+- assertVisible:
+    id: clients-screen
+
+- evalScript: ${output.clientFound = false}
+
+# Type the search
+- tapOn: 'Search by name clients'
```
Comment (mikefeldberg) [inline apps/betterangels/.maestro/lib/screens/clients/search_and_select_client.yml]: Why not use `testId` here? Is it necessary to reference the placeholder text to make sure that maesto taps inside the actual input?

File: apps/betterangels/.maestro/tests/create_delete_client_interaction.yml [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+- assertVisible: 'Profile'
+- assertVisible: 'Back'
+- assertVisible: ${output.fixtureClientName}
+
+# ###############################
+# Go to Interactions Tab ########
+# ###############################
+
+# genearate unique test data
```
Comment (mikefeldberg) [inline apps/betterangels/.maestro/tests/create_delete_client_interaction.yml]: `generate`

File: apps/betterangels/.maestro/tests/create_delete_client_interaction.yml [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+- evalScript: ${output.interactionId = Date.now()}
+- evalScript: ${output.interactionPurpose = "interaction-" + String(output.interactionId)}
+- evalScript: ${output.noteCardTestId = "note-card-" + String(output.interactionPurpose)}
+
+# #########################
+# Add Intgeraction ########
```
Comment (mikefeldberg) [inline apps/betterangels/.maestro/tests/create_delete_client_interaction.yml]: `Interaction`

**Review by mikefeldberg — APPROVED** [review]

File: libs/expo/betterangels/src/lib/screens/Client/Interactions/index.tsx (hunk 2 above)
Comment (tglaz) [reply libs/expo/betterangels/src/lib/screens/Client/Interactions/index.tsx:55]: it's under "interactions" dir so will leave as is

File: apps/betterangels/.maestro/lib/screens/clients/search_and_select_client.yml (hunk 4 above)
Comment (tglaz) [reply apps/betterangels/.maestro/lib/screens/clients/search_and_select_client.yml]: could use either, but this can test what's in the UI.
typically I use testID when there's ambiguity (or maestro just gets confused), but otherwise this works
